site-src/geps/gep-1426.md

```diff
@@ -114,37 +113,44 @@ Services supported as `backendRefs` SHOULD be consistent with expectations for N
     * The namespace colocation requirement effectively forecloses using this strategy to delegate route management, but there is service mesh [user interest](https://github.com/istio/istio/issues/22997) in merging separately-defined routes together to manage large configurations, multiple versions or canary deployments for a single mesh service.
-* NOT already specified as a `backendRef` of any _other_ `HTTPRoute` with a `Service` `parentRef`
+* NOT already specified as a `backendRef` of any _other_ `Route` with a `Service` `parentRef`
    * This restriction is intended to prevent route "inclusion/delegation" patterns or circular references from multiple tiers of routing rules _within mesh configuration_.
-   * A `Service` MAY still be listed as a `backendRef` on a different `HTTPRoute` which only targets a `Gateway` as a `parentRef`. The specific behavior of how Gateways might interact with mesh routing rules will be defined in a future GEP.
-    * A `Service` MAY still be listed as a `backendRef` on the _same_ `HTTPRoute` where it is specified as a `parentRef`, e.g. to only split off some subset of traffic from the default backend endpoints.
+   * A `Service` MAY still be listed as a `backendRef` on a different `Route` which only targets a `Gateway` as a `parentRef`. The specific behavior of how Gateways might interact with mesh routing rules will be defined in a future GEP.
+    * A `Service` MAY still be listed as a `backendRef` on the _same_ `Route` where it is specified as a `parentRef`, e.g. to only split off some subset of traffic from the default backend endpoints.
     * This could be computationally expensive to validate.
 
 This should be enforced by controller validation, which should set an `Accepted: { status: False }` condition with a corresponding new `ParentRefConflict` `RouteConditionReason`.
 
-The logic described here is a bit verbose, but could be validated with conformance tests, and is intended to allow an `HTTPRoute` configuring mesh traffic rules to additionally specify a `Gateway` `parentRef` (or `HTTPRoute` `parentRef` if [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085) moves forward) to apply the same routing rules when exposing a mesh service outside the cluster.
+The logic described here is a bit verbose, but could be validated with conformance tests, and is intended to allow an `Route` configuring mesh traffic rules to additionally specify a `Gateway` `parentRef` (or `Route` `parentRef` if [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085) moves forward) to apply the same routing rules when exposing a mesh service outside the cluster.
+
+### Route types
+
+All types currently defined in the gateway-api core (`HTTP`, `GRPC`, `TCP`, `TLS`, and `UDP`) are available for use in a Mesh implementation.
```

Comment:
Do any reference implementations exist for UDP routes in service mesh, or something approximating a TLS route that differs in behavior from HTTP or TCP routes?

Reply:
TLSRoute largely only makes sense if the parentRef doesn't have a VIP _or_ we allow `hostnames` to be set as well.

UDP - Not sure, perhaps Cilium could.. Istio is planning to but it will be 1-2 years probably still

Reply:
UDP is tricky because it's hard to make sure that the client address is preserved, and most UDP protocols assume that they have the correctly client IP (they don't have mechanisms like x-Forwarded-For in HTTP).

Reply:
I agree but its also orthogonal to using UDPRoute IMO. If a mesh supports UDP, they ought to be able to support UDPRoute as well. Maybe we call it experimental since there are no implementations yet, but the whole thing is experimental for now anyways?

Reply:
Would narrowing the scope to just HTTPRoute, GRPCRoute and TCPRoute lets us at least start making this more general and get this merged, and punt on TLSRoute and UDPRoute until later?

Reply:
That makes sense to me, yeah.